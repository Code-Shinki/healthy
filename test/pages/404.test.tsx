import { cleanup, render } from '@testing-library/react'
import { useRouter } from 'next/router'
import React from 'react'
import Error from '../../src/pages/404'
import { nextRouterMock } from '../mocks/nextRouterMock'

// モック関数を作成
jest.mock('next/router', () => ({ useRouter: jest.fn() }))

describe(`404ページ (/404)`, () => {
  afterEach(cleanup)

  test('スナップショット', async () => {
    // useRouterのメソッドをモック化する
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    const { asFragment } = render(<Error />)
    const tree = asFragment()
    expect(tree).toMatchSnapshot()
  })
})
