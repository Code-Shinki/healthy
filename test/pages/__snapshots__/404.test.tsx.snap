// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404ページ (/404) スナップショット 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="container"
    >
      <h1
        class="title"
      >
        404
      </h1>
      <p
        class="description"
      >
        大変申し訳ございません。
      </p>
      <p
        class="description"
      >
        サーバー側にてエラーが発生いたしました。
      </p>
      <p
        class="description"
      >
        <a
          class="link"
          href="/dashboard"
        >
          ダッシュボード
        </a>
        または
        <a
          class="link"
          href="/"
        >
          トップページ
        </a>
        からアプリケーションにお戻りください。
      </p>
      <p
        class="description"
      >
        お困りの際は
        <a
          class="link"
          href="https://twitter.com/code_shinki"
          rel="noopener noreferrer"
          target="_blank"
        >
          開発者 (Twitter)
        </a>
        へご連絡ください。
      </p>
    </div>
  </div>
</DocumentFragment>
`;
